jmaqs-webservices-jdk8/src/test/java/com/magenic/jmaqs/webservices/jdk8/WebServiceDriverRequestUnitTest/WebServiceDriverPutUnitTest.java

```diff
@@ -0,0 +1,93 @@
+package com.magenic.jmaqs.webservices.jdk8.WebServiceDriverRequestUnitTest;
+
+import com.magenic.jmaqs.utilities.helper.TestCategories;
+import com.magenic.jmaqs.webservices.jdk8.BaseWebServiceTest;
+import com.magenic.jmaqs.webservices.jdk8.WebServiceUtilities;
+import com.magenic.jmaqs.webservices.jdk8.models.Product;
+import java.math.BigDecimal;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.entity.ContentType;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/**
+ * Tests the web service driver Put functionality.
+ */
+public class WebServiceDriverPutUnitTest extends BaseWebServiceTest {
+  /**
+   * verify an object serialized as a XML can do a PUT request.
+   * @throws Exception There was a problem with the test
+   */
+  @Test(groups = TestCategories.WEB_SERVICE)
+  public void webServiceXmlPut() throws Exception {
+    Product p = new Product();
+    p.setCategory("ff");
+    p.setId(4);
+    p.setName("ff");
+    p.setPrice(BigDecimal.valueOf(3.25));
+
+    HttpEntity content = WebServiceUtilities.createStringEntity(p, ContentType.APPLICATION_XML);
+    CloseableHttpResponse response = this.getWebServiceDriver()
+        .putContent("/api/XML_JSON/Put/1", content, ContentType.APPLICATION_XML, true);
+
+    Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
+    Assert.assertEquals(response.getStatusLine().getReasonPhrase(), "OK");
+  }
+
+  /**
+   * verify an object serialized as a JSON can do a PUT request.
+   * @throws Exception There was a problem with the test
+   */
+  @Test(groups = TestCategories.WEB_SERVICE)
+  public void webServiceJsonPut() throws Exception {
+    Product p = new Product();
+    p.setCategory("ff");
+    p.setId(4);
+    p.setName("ff");
+    p.setPrice(BigDecimal.valueOf(3.25));
+
+    HttpEntity content = WebServiceUtilities.createStringEntity(p, ContentType.APPLICATION_JSON);
+    CloseableHttpResponse response = this.getWebServiceDriver()
+        .putContent("/api/XML_JSON/Put/1", content, ContentType.APPLICATION_JSON, true);
+
+    Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
+    Assert.assertEquals(response.getStatusLine().getReasonPhrase(), "OK");
+  }
+
+  /**
+   * verify an object serialized as a String can do a PUT request.
+   * @throws Exception There was a problem with the test
+   */
+  @Test(groups = TestCategories.WEB_SERVICE)
+  public void webServiceStringPut() throws Exception {
+    Product p = new Product();
+    p.setCategory("ff");
+    p.setId(4);
+    p.setName("ff");
+    p.setPrice(BigDecimal.valueOf(3.25));
+
+    HttpEntity content = WebServiceUtilities.createStringEntity(p, ContentType.APPLICATION_JSON);
```

Comment:
Should the Content Type be TEXT_PLAIN here as the test is supposed to serialize as string?

Reply:
TEXT_PLAIN has replaced ContentType.APPLICATION_JSON